### PR TITLE
Added list of book recommendations to home page

### PR DESCRIPTION
**Pull request by jhuseman**

### CODE REVIEW

**Comment by jhuseman** [comment]
This should add book recommendations to the homepage, but currently it's a bit messed up because of having to get the list of recommendations from the heroku app instead of the local server.  Otherwise, I think it should be ready for review.

File: bookfinder/static/js/book-list.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+function add_item_to_list(book, list_id) {
```
Comment (cshinaver) [inline bookfinder/static/js/book-list.js]: Can you convert these tabs to spaces so we are consistent with the other code?

File: bookfinder/static/js/recommendations.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+function get_suggestion_list(user_id, count) {
```
Comment (cshinaver) [inline bookfinder/static/js/recommendations.js]: Also convert tabs to spaces here

File: bookfinder/static/js/search.js [hunk 3]
```diff
@@ -18,31 +18,5 @@ function fill_search(text) {
...
-	var new_isbn_text = document.createTextNode(book.isbn);
-	new_isbn.appendChild(new_isbn_text);
-	new_isbn.setAttribute("class","list-group-item-text");
-	new_info_frame.appendChild(new_isbn);
-	new_entry.appendChild(new_info_frame);
-	new_entry.setAttribute("style","overflow:auto;");
-	new_entry.setAttribute("class","list-group-item");
-	new_entry.setAttribute("href",'/search/prices?isbn=' + book.isbn);
-	document.getElementById("result-list").appendChild(new_entry);
+	add_item_to_list(book,"result-list");
```
Comment (cshinaver) [inline bookfinder/static/js/search.js]: Tabs to spaces

File: bookfinder/templates/index.html [hunk 4]
```diff
@@ -18,5 +18,20 @@
 				</div>
 			</div>
 		</form>
+		<div id="recommendations">
```
Comment (cshinaver) [inline bookfinder/templates/index.html]: Tabs to spaces

File: bookfinder/templates/search/index.html [hunk 5]
```diff
@@ -24,6 +24,7 @@ <h4 class="list-group-item-heading">Please Wait...</h4>
 {% block footer %}
     {{ super() }}
 	<script src="/static/js/get-file.js"></script>
+	<script src="/static/js/book-list.js"></script>
```
Comment (cshinaver) [inline bookfinder/templates/search/index.html]: Tabs to spaces (I see that this file doesn't already have tabs converted to spaces, so converting the entire file would be fine)

**Comment by cshinaver** [comment]
The recommendations seem to not be showing up for me. I'm signed in with a user with `id == 1` and nothing is coming up. I tested the recommendation server with `user_id = 1` and `number_of_preferences = 10` and it worked.

```
$ curl --data "user_id=1&number_of_preferences=10" http://bookfinderrecommender.herokuapp.com/get-recommendation
[{"item_id":7,"value":0.5972222},{"item_id":4,"value":0.5972222},{"item_id":8,"value":0.22222222}]% 
```

File: bookfinder/static/js/book-list.js (hunk 1 above)
Comment (cshinaver) [inline bookfinder/static/js/book-list.js]: These were poorly placed comments on my part. This one's finished. 
(thumbsup)

File: bookfinder/static/js/recommendations.js (hunk 2 above)
Comment (cshinaver) [inline bookfinder/static/js/recommendations.js]: :+1:

**Comment by jhuseman** [comment]
Yes, the recommendations won't show up unless your local database has books with the id numbers 7, 4, and 8 to match the returned suggestions.  This is what I meant by "currently it's a bit messed up" in my first comment. It should work once its on the main server. It just falls back to displaying nothing when a matching book for the id isn't returned.

**Comment by jhuseman** [comment]
I also just realized that I need to make the `Some Recommended books for you:` message disappear when not logged in.  I'll change that right now.

**Comment by jhuseman** [comment]
image
Here's a screenshot of the working suggestions (It's just one book because only one book id matches what's in the returned list).

**Comment by cshinaver** [comment]
The recommended book heading should also not appear when there are no recommendations available.

**Comment by cshinaver** [comment]
Also, are the recommendations getting links for thumbnails to display? The book model does not have a `link` attribute.

**Comment by cshinaver** [comment]
I just added the `thumbnail_link` attribute, so that won't cause an issue anymore. Although I am curious how you were able to load recommendations with a thumbnail without the field being there...

**Comment by cshinaver** [comment]
approved

**Pull request merged**